app/javascript/dashboard/routes/dashboard/settings/customRoles/Index.vue

```diff
@@ -0,0 +1,193 @@
+<script setup>
+import { useAlert } from 'dashboard/composables';
+import SettingsLayout from '../SettingsLayout.vue';
+import BaseSettingsHeader from '../components/BaseSettingsHeader.vue';
+import CustomRoleModal from './component/CustomRoleModal.vue';
+import CustomRoleTableBody from './component/CustomRoleTableBody.vue';
+import CustomRolePaywall from './component/CustomRolePaywall.vue';
+import { computed, onMounted, ref } from 'vue';
+import { useI18n } from 'dashboard/composables/useI18n';
+import { useStore, useMapGetter } from 'dashboard/composables/store';
+
+const store = useStore();
+const { t } = useI18n();
+
+const showCustomRoleModal = ref(false);
+const customRoleModalMode = ref('add');
+const selectedRole = ref(null);
+const loading = ref({});
+const showDeleteConfirmationPopup = ref(false);
+const activeResponse = ref({});
+const customRoleAPI = ref({ message: '' });
+
+const records = useMapGetter('customRole/getCustomRoles');
+const uiFlags = useMapGetter('customRole/getUIFlags');
+
+const deleteConfirmText = computed(
+  () => `${t('CUSTOM_ROLE.DELETE.CONFIRM.YES')} ${activeResponse.value.name}`
+);
+
+const deleteRejectText = computed(
+  () => `${t('CUSTOM_ROLE.DELETE.CONFIRM.NO')} ${activeResponse.value.name}`
+);
+
+const deleteMessage = computed(() => {
+  return ` ${activeResponse.value.name} ? `;
+});
+
+const isFeatureEnabledOnAccount = useMapGetter(
+  'accounts/isFeatureEnabledonAccount'
+);
+
+const currentAccountId = useMapGetter('getCurrentAccountId');
+
+const isBehindAPaywall = computed(() => {
+  return !isFeatureEnabledOnAccount.value(
+    currentAccountId.value,
+    'custom_roles'
+  );
+});
+
+const fetchCustomRoles = async () => {
+  try {
+    await store.dispatch('customRole/getCustomRole');
+  } catch (error) {
+    // Ignore Error
+  }
+};
+
+onMounted(() => {
+  fetchCustomRoles();
+});
+
+const showAlertMessage = message => {
+  loading[activeResponse.value.id] = false;
```

Comment:
Should we check loading.value here. Please check this condition.
```suggestion
loading.value[activeResponse.value.id] = false;
```